core/alsp_src/builtins/blt_shl.pro

```diff
@@ -53,21 +55,48 @@ start_shell0(DefaultShellCall)
 	output_system_banner(CLInfo),
 
 	library_setup,
-/* WHY IS THIS MISSING?
-#if (all_procedures(syscfg,intconstr,0,_))
-	rel_arith:set_ics(cs(0,0,0)),
-#endif
-*/
+
 	(clause(dvf,_) -> qkc ; true),
-%	dvf,
 	load_cl_files(CLInfo),
 	process_cl_asserts(CLInfo),
 	!,
-		%% 
 	ss_load_dot_alspro(CLInfo),
+
+	HistoryFile = '.alspro_history',
+
+	check_setup_history_file(HistoryFile),
+	check_load_prev_history,
+
 	setup_init_goal(CLInfo, ShellCall),
 	user:ShellCall.
 
+history_file_locn(L) :- user:history_file_locn(L), !.
+history_file_locn(home).
+
+check_setup_history_file(HistoryFile) :-
+   history_file_locn(HFL),
+   setup_history_file(HFL, HistoryFile).
+
+setup_history_file(home, HistoryFile)
+	:-!,
+	getenv('HOME', UserHomePath),
+	pathPlusFile(UserHomePath, HistoryFile, PathToHistoryFile),
+	sio_set_history_file(PathToHistoryFile).
+
+setup_history_file(local, HistoryFile)
+	:-!,
+	sio_set_history_file(HistoryFile).
+
+setup_history_file(_).
+
+check_load_prev_history
+	:-
+	user:no_load_prev_history,
+	!,
+	sio_set_load_prev_history.
```

Comment:
> On Sep 13, 2016, at 5:04 PM, Chuck Houpt <notifications@github.com> wrote:
> 
> In core/alsp_src/builtins/blt_shl.pro:
> 
> > +	:-!,
> > +	getenv('HOME', UserHomePath),
> > +	pathPlusFile(UserHomePath, HistoryFile, PathToHistoryFile),
> > +	sio_set_history_file(PathToHistoryFile).
> > +
> > +setup_history_file(local, HistoryFile)
> > +	:-!,
> > +	sio_set_history_file(HistoryFile).
> > +
> > +setup_history_file(_).
> > +
> > +check_load_prev_history
> > +	:-
> > +	user:no_load_prev_history,
> > +	!,
> > +	sio_set_load_prev_history.
> 
> I think sio_set_load_prev_history/0 should be renamed to reflect that it does the exact opposite.

You’re right.

> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
>